test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -223,6 +230,129 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+		if !exists {
+			return nil, fmt.Errorf("network name not found for interface %s", iface)
+		}
+		netAttachDef, err := client.K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Get(context.TODO(), networkName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to get NetworkAttachmentDefinition %s for interface %s: %w", networkName, iface, err)
+		}
+
+		var netAttachDefConfig struct {
+			IPAM struct {
+				IPPools []string `json:"ippools"`
+			} `json:"ipam"`
+		}
+
+		if err := json.Unmarshal([]byte(netAttachDef.Spec.Config), &netAttachDefConfig); err != nil {
+			return nil, fmt.Errorf("failed to parse NetworkAttachmentDefinition config JSON for interface %s: %w", iface, err)
+		}
+
+		ipPoolsMap[iface] = netAttachDefConfig.IPAM.IPPools
+
+	}
+	return ipPoolsMap, nil
+}
+
+func (data *testData) checkIPReleased(ipPools map[string][]string, ifacesIPs map[string]net.IP, ifaces []string) error {
+	crdClient := data.e2eTestData.CRDClient
+
+	return wait.PollUntilContextTimeout(context.Background(), time.Second, 10*time.Second, true, func(ctx context.Context) (bool, error) {
+		for _, iface := range ifaces {
+			ipPoolName, exists := ipPools[iface]
+			if !exists {
+				return false, fmt.Errorf("no IPPool found for interface %s", iface)
+			}
+
+			podIP, exists := ifacesIPs[iface]
+			if !exists {
+				return false, fmt.Errorf("no IP found for interface %s", iface)
+			}
+
+			ipPool, err := crdClient.CrdV1beta1().IPPools().Get(ctx, ipPoolName[0], metav1.GetOptions{})
+			if err != nil {
+				return false, fmt.Errorf("failed to get IPPool %s: %w", ipPoolName[0], err)
+			}
+
+			for _, ipAddress := range ipPool.Status.IPAddresses {
+				if podIP.String() == ipAddress.IPAddress {
+					return false, nil
+				}
+			}
+			logs.Infof("Released PodIP: %v, Interface: %s", podIP, iface)
+		}
+		return true, nil
+	})
+}
+
+func (data *testData) getOVSPortsOnSecondaryBridge(t *testing.T, nodeName string) ([]string, error) {
+	cmd := []string{"ovs-vsctl", "list-ports", secondaryOVSBridge}
+
+	stdout, stderr, err := data.e2eTestData.RunCommandFromAntreaPodOnNode(nodeName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to run ovs-vsctl on node %s: %v\nstderr: %s", nodeName, err, stderr)
+	}
+
+	ovsPorts := strings.Fields(stdout)
+	return ovsPorts, nil
+}
+
+// reconcilationAfterAgentRestart verifies OVS Ports cleanup and IP release.
+func (data *testData) reconcilationAfterAgentRestart(t *testing.T) {
+	for _, pod := range data.pods {
+		ovsPortsBeforeAgentRestart, err := data.getOVSPortsOnSecondaryBridge(t, pod.nodeName)
+		require.NoError(t, err, "Failed to get Secondary bridge OVS Ports before agent restart")
+
+		beforeAgentRestartIPsAndIfaces, err := data.listPodIPs(pod)
+		require.NoError(t, err, "Failed to get Pod IP before agent restart")
+
+		require.NoError(t, data.e2eTestData.RestartAntreaAgentPods(30*time.Second), "Failed to restart Antrea agent Pods")
```

Comment:
please update the test so that you only restart the Agents once:

1) collect port data and IPs first
1) restart Agents
1) collect new data, compare